### ISSUE TITLE
virtualbox vbox additions broken

### ISSUE THREAD

**Issue by stefco** [stefco/geco_vm#25]
likely due to kernel updates happening after install; try to fix by installing everything before installing virtualbox additions

**Comment by stefco** [comment]
seems like this guy installs `dkms` before the guest additions to deal with the kernel issue:

https://github.com/shiguredo/packer-templates/blob/develop/ubuntu-12.04/scripts/virtualbox.sh

**Comment by stefco** [comment]
good suggestions here: http://askubuntu.com/questions/98416/error-kernel-headers-not-found-but-they-are-in-place

**Comment by stefco** [comment]
headers are definitely missing, see if it gets fixed with a [`dkms`](https://en.wikipedia.org/wiki/Dynamic_Kernel_Module_Support) install:

```
==> virtualbox-iso: Provisioning with shell script: scripts/common/vmtools.sh
    virtualbox-iso: + HOME_DIR=/home/vagrant
    virtualbox-iso: + mkdir -p /tmp/vbox
    virtualbox-iso: + cat /home/vagrant/.vbox_version
    virtualbox-iso: + ver=5.0.16
    virtualbox-iso: + mount -o loop /home/vagrant/VBoxGuestAdditions_5.0.16.iso /tmp/vbox
    virtualbox-iso: mount: block device /home/vagrant/VBoxGuestAdditions_5.0.16.iso is write-protected, mounting read-only
    virtualbox-iso: + sh /tmp/vbox/VBoxLinuxAdditions.run
    virtualbox-iso: Verifying archive integrity... All good.
    virtualbox-iso: Uncompressing VirtualBox 5.0.16 Guest Additions for Linux............
    virtualbox-iso: VirtualBox Guest Additions installer
    virtualbox-iso: Copying additional installer modules ...
    virtualbox-iso: Installing additional modules ...
    virtualbox-iso: Removing existing VirtualBox non-DKMS kernel modules ...done.
    virtualbox-iso: Building the VirtualBox Guest Additions kernel modules
    virtualbox-iso: The headers for the current running kernel were not found. If the following
    virtualbox-iso: module compilation fails then this could be the reason.
    virtualbox-iso:
    virtualbox-iso: Building the main Guest Additions module ...done.
    virtualbox-iso: Building the shared folder support module ...done.
    virtualbox-iso: Building the OpenGL support module ...done.
    virtualbox-iso: Doing non-kernel setup of the Guest Additions ...done.
    virtualbox-iso: Starting the VirtualBox Guest Additions ...done.
    virtualbox-iso: Installing the Window System drivers
    virtualbox-iso: Could not find the X.Org or XFree86 Window System, skipping.
    virtualbox-iso: + echo VBoxLinuxAdditions.run exited 1 and is suppressed. For more read https://www.virtualbox.org/ticket/12479
    virtualbox-iso: VBoxLinuxAdditions.run exited 1 and is suppressed. For more read https://www.virtualbox.org/ticket/12479
    virtualbox-iso: + umount /tmp/vbox
    virtualbox-iso: + rm -rf /tmp/vbox
    virtualbox-iso: + rm -f /home/vagrant/VBoxGuestAdditions_5.0.16.iso
```

**Comment by stefco** [comment]
`dkms` did not help. its effect:

```
==> virtualbox-iso: Provisioning with shell script: scripts/common/vmtools.sh
    virtualbox-iso: + HOME_DIR=/home/vagrant
    virtualbox-iso: + apt-get -y install dkms
    virtualbox-iso: Reading package lists...
    virtualbox-iso: Building dependency tree...
    virtualbox-iso: Reading state information...
    virtualbox-iso: The following NEW packages will be installed:
    virtualbox-iso: dkms
    virtualbox-iso: 0 upgraded, 1 newly installed, 0 to remove and 7 not upgraded.
    virtualbox-iso: Need to get 73.3 kB of archives.
    virtualbox-iso: After this operation, 350 kB of additional disk space will be used.
    virtualbox-iso: Get:1 http://us.archive.ubuntu.com/ubuntu/ precise-updates/main dkms all 2.2.0.3-1ubuntu3.6 [73.3 kB]
    virtualbox-iso: debconf: unable to initialize frontend: Dialog
    virtualbox-iso: debconf: (Dialog frontend will not work on a dumb terminal, an emacs shell buffer, or without a controlling terminal.)
    virtualbox-iso: debconf: falling back to frontend: Readline
    virtualbox-iso: debconf: unable to initialize frontend: Readline
    virtualbox-iso: debconf: (This frontend requires a controlling tty.)
    virtualbox-iso: debconf: falling back to frontend: Teletype
    virtualbox-iso: dpkg-preconfigure: unable to re-open stdin:
    virtualbox-iso: Fetched 73.3 kB in 1s (59.7 kB/s)
    virtualbox-iso: Selecting previously unselected package dkms.
    virtualbox-iso: (Reading database ... 131782 files and directories currently installed.)
    virtualbox-iso: Unpacking dkms (from .../dkms_2.2.0.3-1ubuntu3.6_all.deb) ...
    virtualbox-iso: Processing triggers for man-db ...
    virtualbox-iso: Setting up dkms (2.2.0.3-1ubuntu3.6) ...
    virtualbox-iso: + mkdir -p /tmp/vbox
    virtualbox-iso: + cat /home/vagrant/.vbox_version
    virtualbox-iso: + ver=5.0.16
    virtualbox-iso: + mount -o loop /home/vagrant/VBoxGuestAdditions_5.0.16.iso /tmp/vbox
    virtualbox-iso: mount: block device /home/vagrant/VBoxGuestAdditions_5.0.16.iso is write-protected, mounting read-only
    virtualbox-iso: + sh /tmp/vbox/VBoxLinuxAdditions.run
    virtualbox-iso: Verifying archive integrity... All good.
    virtualbox-iso: Uncompressing VirtualBox 5.0.16 Guest Additions for Linux............
    virtualbox-iso: VirtualBox Guest Additions installer
    virtualbox-iso: Copying additional installer modules ...
    virtualbox-iso: Installing additional modules ...
    virtualbox-iso: Removing existing VirtualBox DKMS kernel modules ...done.
    virtualbox-iso: Removing existing VirtualBox non-DKMS kernel modules ...done.
    virtualbox-iso: Building the VirtualBox Guest Additions kernel modules ...done.
    virtualbox-iso: Doing non-kernel setup of the Guest Additions ...done.
    virtualbox-iso: Starting the VirtualBox Guest Additions ...done.
    virtualbox-iso: Installing the Window System drivers
    virtualbox-iso: Could not find the X.Org or XFree86 Window System, skipping.
    virtualbox-iso: + echo VBoxLinuxAdditions.run exited 1 and is suppressed. For more read https://www.virtualbox.org/ticket/12479
    virtualbox-iso: VBoxLinuxAdditions.run exited 1 and is suppressed. For more read https://www.virtualbox.org/ticket/12479
    virtualbox-iso: + umount /tmp/vbox
    virtualbox-iso: + rm -rf /tmp/vbox
    virtualbox-iso: + rm -f /home/vagrant/VBoxGuestAdditions_5.0.16.iso
```

**Comment by stefco** [comment]
Try:
- [ ] change template back to old name
- [x] put cleanup script back in
- [ ] see when provision.sh breaks it

**Comment by stefco** [comment]
the cleanup script ended up being the missing piece. fixed in 4207be8
